script.js

```diff
@@ -124,21 +143,38 @@ window.onload = function() {
         exposureTimeInput.value = exposureTimeText.replace('s ', '');
         isoSpeedRatingsInput.value = isoSpeedRatingsText.replace('ISO', '');
 
-        // テキストを描画
-        let textHeight = finalText ? textCenter - lineSpacing : textCenter + baseFontSize / 2; // 2行目テキストがある場合は上に、ない場合は中央にずらす
-        ctx.fillText(text1, textStart, textHeight);
-        ctx.font = '900 ' + baseFontSize + 'px sans-serif';  // フォントの設定を変更
-        ctx.fillStyle = '#000000';  // 文字色
-        ctx.fillText(text2, textStart + text1Width, textHeight);
-        ctx.font = '600 ' + baseFontSize + 'px sans-serif';  // フォントの設定を戻す
-        ctx.fillText(text3, textStart + text1Width + text2Width, textHeight);
-
-        ctx.textAlign = 'center';  // 水平中央揃え
-        ctx.font = '400 ' + baseFontSize * 0.8 + 'px sans-serif';  // フォントの設定
-        ctx.fillStyle = '#747474';  // 文字色
-        ctx.fillText(finalText, canvas.width / 2, textCenter + lineSpacing + baseFontSize);
+        // テキスト描画の関数
+        function renderTexts() {
+            let textHeight = finalText ? textCenter - lineSpacing : textCenter + baseFontSize / 2; // 2行目テキストがある場合は上に、ない場合は中央にずらす
+            ctx.fillText(text1, textStart, textHeight);
+            ctx.font = '700 ' + baseFontSize + 'px ' + fontFamily;  // フォントの設定を変更
+            ctx.fillStyle = '#000000';  // 文字色
+            ctx.fillText(text2, textStart + text1Width, textHeight);
+            ctx.font = '500 ' + baseFontSize + 'px ' + fontFamily;  // フォントの設定を戻す
+            ctx.fillStyle = '#343434';  // 文字色
+            ctx.fillText(text3, textStart + text1Width + text2Width, textHeight);
+
+            ctx.textAlign = 'center';  // 水平中央揃え
+            ctx.font = '400 ' + baseFontSize * 0.8 + 'px ' + fontFamily;  // フォントの設定
+            ctx.fillStyle = '#747474';  // 文字色
+            ctx.fillText(finalText, canvas.width / 2, textCenter + lineSpacing + baseFontSize);
+        }
+
+        // 画像の描画処理
+        function showImage() {
+            resultImage.src = canvas.toDataURL(); 
+        }
 
-        resultImage.src = canvas.toDataURL();
+        // フォント読み込みとテキストの描画（フォントの二重読み込み防止処理）
```

Comment:
Interをロードする前にはすでにsans-serifでテキスト座標を計算していることが原因のようです（ctx.measureText().widthが変わる）

```suggestion
       let text1Width = ctx.measureText(text1).width;
```

Reply:
draw関数に対してフォントのロードを行うといいと思います！

```
// フォント読み込みとテキストの描画（フォントの二重読み込み防止処理）
if (isFontsLoaded) {
  draw(exifData);
} else {
  loadFonts().then(function() {
      draw(exifData);
  });
}
```

Reply:
最高！承知です！

Reply:
:igyo: